numbers.go

```diff
@@ -1,121 +1,56 @@
 // a Q number. The key benefit of this is in valueMatcher.transitionOn, which incurs the cost of
 // making a Q number only if it is known that the valueMatcher's NFA can benefit from it and
 // that the number in the incoming event can in fact be made a Q number.
 
-const (
-	TenE6               = 1e6
-	FiveBillion         = 5e9
-	Hexes               = "0123456789ABCDEF"
-	MaxFractionalDigits = 5
-)
-
 type qNumber []byte
 
 // qNumFromBytes works out whether a string representing a number falls within the
 // limits imposed for Q numbers. It is heavily optimized and relies on  the form
 // of the number already having been validated, e.g. by flattenJSON().
 func qNumFromBytes(bytes []byte) (qNumber, error) {
-	// shortcut: The shorest number with more than 5 fractional digits is like 0.123456
-	if len(bytes) < 8 {
-		numeric, err := strconv.ParseFloat(string(bytes), 64)
-		if err != nil {
-			return nil, errors.New("not a float") // should never happen, json parser upstream
-		}
-		return qNumFromFloat(numeric)
-	}
-	// compute number of fractional digits. The loop below relies on the fact that anything between '.' and either
-	// 'e' or the end of the string must be a digit, as must anything between 'e' and the end of the string.
-	//. NOTE: This will be fooled by "35.000000"
-	fracStart := 0
-	expStart := 0
-	index := 0
-	var utf8Byte byte
-	fractionalDigits := 0
-ForEachByte:
-	for index, utf8Byte = range bytes {
-		switch utf8Byte {
-		case '.':
-			fracStart = index + 1
-		case 'e', 'E':
-			expStart = index + 1
-			break ForEachByte
-		}
-	}
-	if fracStart != 0 {
-		fractionalDigits = index - fracStart
-	}
-	// if too many fractional digits, perhaps the exponent will push the '.' to the right
-	if fractionalDigits > MaxFractionalDigits {
-		if expStart != 0 {
-			exp, err := strconv.ParseInt(string(bytes[expStart:]), 10, 32)
-			if err == nil {
-				fractionalDigits -= int(exp)
-			}
-		}
-	}
-	if fractionalDigits > MaxFractionalDigits {
-		return nil, errors.New("more than 5 fractional digits")
-	}
-
 	numeric, err := strconv.ParseFloat(string(bytes), 64)
 	if err != nil {
-		return nil, errors.New("not a float") // shouldn't happen, upstream parser should prvent
+		return nil, errors.New("not a float") // should never happen, json parser upstream
 	}
-	return qNumFromFloat(numeric)
+	return qNumFromFloat(numeric), nil
 }
 
-func qNumFromFloat(f float64) (qNumber, error) {
-	if f < -FiveBillion || f > FiveBillion {
-		return nil, errors.New("value must be between -5e9 and +5e9 inclusive")
-	}
-	value := uint64(TenE6 * (FiveBillion + f))
-	return toHexStringSkippingFirstByte(value), nil
+// qNumFromFLoat is here mostly to support testing
+func qNumFromFloat(f float64) qNumber {
+	return numbitsFromFloat64(f).toUTF8()
 }
 
-func toHexStringSkippingFirstByte(value uint64) []byte {
-	var buf [8]byte
-	binary.BigEndian.PutUint64(buf[:], value)
-	var outputChars [14]byte
-	for i, utf8Byte := range buf {
-		if i == 0 {
-			continue
+/*
```

Comment:
drop?

Reply:
I want to keep this around because it's very useful for debugging and I will definitely need it when I'm implementing the range construct.  I guess I should un-comment and use in a test so that I don't get lint complaints about unused function.